hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflowModule.java

```diff
@@ -87,4 +88,9 @@ static Supplier<AutoCloseableWrapper<HederaState>> provideStateSupplier(@NonNull
     static LongSupplier provideConsensusSecond(@NonNull final TransactionContext txnCtx) {
         return () -> txnCtx.consensusTime().getEpochSecond();
     }
+
+    @Provides
+    static ReadableStoreFactory provideReadableStoreFactory(final Supplier<AutoCloseableWrapper<HederaState>> state) {
+        return new ReadableStoreFactory(state.get().get());
```

Comment:
Which state is used when a `ReadableStoreFactory` is created like this?

Reply:
Since we are passing in `Supplier<AutoCloseableWrapper<HederaState>>` it uses `platform.getLatestImmutableState` which is provided as 
```
  @Provides
    @SuppressWarnings({"unchecked", "rawtypes"})
    static Supplier<AutoCloseableWrapper<HederaState>> provideStateSupplier(@NonNull final Platform platform) {
        // Always return the latest immutable state until we support state proofs
        return () -> (AutoCloseableWrapper) platform.getLatestImmutableState(HandleWorkflowModule.class.getName());
    }
```